apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/rest/endpoint/NetworkEndpointTest.java

```diff
@@ -0,0 +1,318 @@
+/*
+ *
+ *  Copyright © 2018-2019 Apollo Foundation
+ *
+ */
+
+package com.apollocurrency.aplwallet.apl.core.rest.endpoint;
+
+import com.apollocurrency.aplwallet.api.dto.PeerDTO;
+import com.apollocurrency.aplwallet.apl.core.peer.Peer;
+import com.apollocurrency.aplwallet.apl.core.rest.converter.PeerConverter;
+import com.apollocurrency.aplwallet.apl.core.rest.service.NetworkService;
+import com.apollocurrency.aplwallet.apl.testutil.EntityProducer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.jboss.resteasy.core.Dispatcher;
+import org.jboss.resteasy.mock.MockDispatcherFactory;
+import org.jboss.resteasy.mock.MockHttpRequest;
+import org.jboss.resteasy.mock.MockHttpResponse;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.jboss.resteasy.mock.MockHttpRequest.get;
+import static org.jboss.resteasy.mock.MockHttpRequest.post;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class NetworkEndpointTest {
+    private static final String PEER_ADDRESS = "192.168.2.68";
+    private static final String WRONG_PEER_ADDRESS = "10.0.0.1";
+    private static final String ANNOUNCED_ADDRESS = "10.10.10.10";
+
+    private static ObjectMapper mapper = new ObjectMapper();
+    private static Dispatcher dispatcher;
+
+    @BeforeAll
+    static void setupClass(){
```

Comment:
Setting up mocks for all test methods make it painful to determine, which mock are related for specific test methods. In the 'given' section of each test method you can setup needed mocks .

Reply:
Yes, sure